sdk/src/test/java/com/horizen/account/transaction/EthereumTransactionTest.java

```diff
@@ -157,7 +206,18 @@ public void ethereumEIP1559RawTransactionTest() throws TransactionSemanticValidi
     }
 
     @Test
-    public void ethereumSignedRawTransactionTest() throws TransactionSemanticValidityException {
+    public void ethereumSignedRawTransactionTest() {
+        // Test 0: direct constructor
+        try {
+            var someTx = new EthereumTransaction(
+                    "0x", someValue, someValue, someValue,
+                    someValue, "", signedRawTransaction.getSignatureData()
+            );
+            assertNotNull(someTx.getSignature());
+        } catch (Exception ignored) {
+            fail("Should not happen");
```

Comment:
Hm, if an exception means the test failed, I don't think you need or should catch it at all. Just let it throw and the test will fail. If an exception is expected for the test to succeed I have been using `assertThrows()`.

@i-Alex What do you think is best practice here?

Reply:
`assertThrows` is good. It will allow to fail the test with a proper message and result structure.